digest.go

```diff
@@ -84,17 +86,26 @@ func (a *DigestAuth) Purge(count int) {
  (or requires reauthentication).
 */
 func (a *DigestAuth) RequireAuth(w http.ResponseWriter, r *http.Request) {
+	a.mutex.RLock()
 	if len(a.clients) > a.ClientCacheSize+a.ClientCacheTolerance {
 		a.Purge(a.ClientCacheTolerance * 2)
 	}
+	a.mutex.RUnlock()
+
 	nonce := RandomKey()
+
+	a.mutex.Lock()
 	a.clients[nonce] = &digest_client{nc: 0, last_seen: time.Now().UnixNano()}
+	a.mutex.Unlock()
+
+	a.mutex.RLock()
```

Comment:
In case the mutex only protects `a.clients` I don't see any access to it in this part.